### PR TITLE
feat(tap): add more diagnostics to tap reporting

### PR DESCRIPTION
**Pull request by lkipke**
This adds some more diagnostics to our TAP reporting, so that it's more clear what errors occurred, and what the stack trace was for those errors.

### CODE REVIEW

File: resources/assert_lib.brs [hunk 1]
```diff
@@ -17,65 +17,90 @@ sub __equal(actual, expected, error)
...
-    return {
-        error: {
-            message: errMessage
-        },
-        stack: {}
-    }
+function __formatError(error)
+    ' Get the stack trace where the failed test is, filtering out any roca frames
+    fileFilters = ["roca"]
+    numStackFrames = 2
```
Comment (sjbarag) [inline resources/assert_lib.brs:84]: I wouldn't be opposed to `3` here? 🤷

File: resources/assert_lib.brs [hunk 2]
```diff
@@ -17,65 +17,90 @@ sub __equal(actual, expected, error)
...
+    return error
 end function
 
 sub __deepEquals(actual, expected, error)
     if __roca_deepEquals(actual, expected) = true then
         m.__pass()
     else
-        m.__fail(m.formatError(error))
+        m.__fail(m.formatError({
+            message: error
```
Comment (sjbarag) [inline resources/assert_lib.brs:95]: Should we add `funcName: "m.assert.deepEquals` to this object?

**Review by sjbarag — APPROVED** [review]
Looks great!  I'd love to see an example of the output if you've got one 😄

File: resources/assert_lib.brs (hunk 2 above)
Comment (lkipke) [reply resources/assert_lib.brs:95]: great catch!

File: resources/assert_lib.brs (hunk 1 above)
Comment (lkipke) [reply resources/assert_lib.brs:84]: sure thing!

**Comment by lkipke** [comment]
Example output:

Test case:
```brs
function main(args as object)
    return roca(args).describe("Child1", sub()
        m.describe("initialization", sub()
            m.it("initializes without error", sub()
                m.assert.equal("abcd", "abce", "they are not equal")
            end sub)
        end sub)
    end sub)
end function
```

TAP output:
```yaml
TAP version 13
1..1
# Subtest: Child1
    1..1
    # Subtest: initialization
        1..1
        not ok 1 - initializes without error
            ---
                actual: abcd
                expected: abce
                funcname: m.assert.equal
                message: they are not equal
                stack: /Users/levi.kipke/Projects/roku-hello-world/tests/Child1.test.brs:5:16
                stackframes:
                    - /Users/levi.kipke/Projects/roku-hello-world/tests/Child1.test.brs:5:16
                    - /Users/levi.kipke/Projects/roku-hello-world/tests/Child1.test.brs:2:11
                    - /Users/levi.kipke/Projects/roku-hello-world/tests/Child1.test.brs:2:4
            ...
    not ok 1 - initialization
ok 1 - Child1
```

"spec" output
```
Child1
  initialization
    1) initializes without error

  0 passing (38ms)
  1 failing

  1) Child1 initialization initializes without error:
     Error: initializes without error
      at tests/Child1.test.brs:5:16

error Command failed with exit code 1.
```

**Pull request merged**